interface/grpc/core/deploy.go

```diff
@@ -29,15 +34,53 @@ func (s *Server) DeployService(stream coreapi.Core_DeployServiceServer) error {
 		err             error
 	)
 
-	sr := newDeployServiceStreamReader(stream)
-	url, err := sr.GetURL()
-	if err != nil {
+	// receive the first message in the stream.
+	if err := sr.RecvMessage(); err != nil {
 		return err
 	}
-	if url != "" {
-		service, validationError, err = s.api.DeployServiceFromURL(url, api.DeployServiceStatusOption(statuses))
+
+	deployOptions := []api.DeployServiceOption{
+		api.DeployServiceStatusOption(statuses),
+	}
+
+	// receive confirmation for a force deploy.
```

Comment:
no tests for new behavior.

Reply:
we don't have a way of faking _api_ pkg yet. feel free to push some integration tests.

Reply:
But I'm not taking about api, I'm talking about grpc only - L47-L70 - you don't need api to test this

Reply:
Could you please provide an example, I wonder how it can be tested without faking _api_ calls.

Reply:
I don't think that would be a proper way of testing the DeployService. Feel free to create an another PR based on this one to implement the tests. So, we can compare and merge it to this PR if it's convenient enough.